lib/ansible/modules/storage/hpe/hpe3par_cpg.py

```diff
@@ -0,0 +1,435 @@
+      - TiB
+    default: GiB
+    description:
+      - "Unit of growth increment."
+    required: false
+  growth_limit:
+    default: -1.0
+    description:
+      - "Specifies that the autogrow operation is limited to the specified
+       storage amount that sets the growth limit.\n"
+    required: false
+  growth_limit_unit:
+    choices:
+      - MiB
+      - GiB
+      - TiB
+    default: GiB
+    description:
+      - "Unit of growth limit."
+    required: false
+  growth_warning:
+    default: -1.0
+    description:
+      - "Specifies that the threshold of used logical disk space when exceeded
+       results in a warning alert.\n"
+    required: false
+  growth_warning_unit:
+    choices:
+      - MiB
+      - GiB
+      - TiB
+    default: GiB
+    description:
+      - "Unit of growth warning."
+    required: false
+  high_availability:
+    choices:
+      - PORT
+      - CAGE
+      - MAG
+    description:
+      - "Specifies that the layout must support the failure of one port pair,
+       one cage, or one magazine.\n"
+    required: false
+  raid_type:
+    choices:
+      - R0
+      - R1
+      - R5
+      - R6
+    description:
+      - "Specifies the RAID type for the logical disk."
+    required: false
+  set_size:
+    default: -1
+    description:
+      - "Specifies the set size in the number of chunklets."
+    required: false
+  state:
+    choices:
+      - present
+      - absent
+    description:
+      - "Whether the specified CPG should exist or not."
+    required: true
+extends_documentation_fragment: hpe3par
+short_description: "Manage HPE 3PAR CPG"
+version_added: "2.6"
+'''
+
+
+EXAMPLES = r'''
+    - name: Create CPG "{{ cpg_name }}"
+      hpe3par_cpg:
+        storage_system_ip="{{ storage_system_ip }}"
+        storage_system_username="{{ storage_system_username }}"
+        storage_system_password="{{ storage_system_password }}"
+        state=present
+        cpg_name="{{ cpg_name }}"
+        domain="{{ domain }}"
+        growth_increment="{{ growth_increment }}"
+        growth_increment_unit="{{ growth_increment_unit }}"
+        growth_limit="{{ growth_limit }}"
+        growth_limit_unit="{{ growth_limit_unit }}"
+        growth_warning="{{ growth_warning }}"
+        growth_warning_unit="{{ growth_warning_unit }}"
+        raid_type="{{ raid_type }}"
+        set_size="{{ set_size }}"
+        high_availability="{{ high_availability }}"
+        disk_type="{{ disk_type }}"
+
+    - name: Delete CPG "{{ cpg_name }}"
+      hpe3par_cpg:
+        storage_system_ip="{{ storage_system_ip }}"
+        storage_system_username="{{ storage_system_username }}"
+        storage_system_password="{{ storage_system_password }}"
+        state=absent
+        cpg_name="{{ cpg_name }}"
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from hpe3par_sdk import client
+except ImportError:
+    client = None
+
+
+def convert_to_binary_multiple(size, size_unit):
```

Comment:
We have this functionality already in Ansible.
Personally I would make the interface support value + unit, rather than 2 parameters.

Reply:
Modified to reuse the functionality from module_utils